pageserver/src/tenant/timeline.rs

```diff
@@ -1697,7 +1697,8 @@ impl Timeline {
                         Discovered::Temporary(name) => (name, "temporary timeline file"),
                         Discovered::TemporaryDownload(name) => (name, "temporary download"),
                     };
-                    path.push(name);
+                    let name_path = name.to_str().map(Utf8Path::new).expect("non-Unicode path");
```

Comment:
If `Timeline::init::scan_timeline_dir` would iterate over the directory with `Utf8Path::read_dir_utf8` and have `enum Decision` carry `String` instead of `OsString` this would be statically avoided.